src/main/groovy/com/bmuschko/gradle/docker/tasks/image/DockerBuildImage.groovy

```diff
@@ -124,13 +127,13 @@ class DockerBuildImage extends AbstractDockerRemoteApiTask implements RegistryCr
     }
 
     void saveImageId() {
-        TaskStateHelper state = new TaskStateHelper(name, project.buildDir)
+        TaskStateHelper state = new TaskStateHelper(this.class.simpleName, project.buildDir)
         state.put("imageId", getImageId())
     }
 
     boolean previouslyBuiltImageExists() {
         boolean imageExists = false
-        String prevImageId = new TaskStateHelper(name, project.buildDir).get("imageId")
+        String prevImageId = new TaskStateHelper(this.class.simpleName, project.buildDir).get("imageId")
```

Comment:
TaskStateHelper had to be changed to work off of the class name itself, which doesn't change, rather than the potentially changing name of the user defined task (i.e. task helloWorld(type: DockerBuildImage), task helloWorld2(type: DockerBuildImage)).